runtime/shared/src/main/scala/org/http4s/googleapis/runtime/auth/IdentityPoolCredentials.scala

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2024 Christopher Davenport
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package googleapis.runtime.auth
+
+import cats.effect.Temporal
+import cats.syntax.all._
+
+import client.Client
+import io.circe.JsonObject
+import io.circe.Json
+import org.http4s.circe.jsonEncoderOf
+import org.http4s.headers.Authorization
+import CredentialsFile.ExternalAccount.ExternalCredentialSource
+import fs2.io.file.Files
+import fs2.io.file.Path
+import fs2.io.IOException
+import cats.effect.Concurrent
+
+import io.circe.Decoder
+import io.circe.generic.semiauto.deriveDecoder
+import org.http4s.circe.jsonOf
+
+object IdentityPoolCredentials
```

Comment:
Each platform has different way(s) of retrieving subject token. These retrievals are extracted into

- ExternalAccountSubjectTokenProvider for local file and remote url  subject token
- AwsSubjectTokenProvider for AWS subject token